src/cargo/core/resolver/types.rs

```diff
@@ -89,26 +89,26 @@ impl ResolverProgress {
     }
 }
 
-#[derive(Clone, Copy)]
```

Comment:
This seems like it's somewhat of a loss, but is there a reason we couldn't use `&'a BTreeSet` here for the `Required` list?

Reply:
I don't see how. (love to be proven wrong.) This change was to get rid of the lifetime, so that it could be used as the key in the `DepsCache` above. Then I switched from a `Vec` to a `BTreeSet` to reduce the number of equivalent configurations that can end up in the cache. Then added the `Rc` as that was the kind of reference I usually had.

This explanation should be a comment. I will add after #6860 lands.